src/util/mod.rs

```diff
@@ -316,12 +316,21 @@ impl<'a, T: 'a, V: 'a, E> TryInsertWith<'a, V, E> for Entry<'a, T, V> {
 }
 
 /// Convert Duration to milliseconds.
+#[inline]
 pub fn duration_to_ms(d: Duration) -> u64 {
     let nanos = d.subsec_nanos() as u64;
     // Most of case, we can't have so large Duration, so here just panic if overflow now.
     d.as_secs() * 1_000 + (nanos / 1_000_000)
 }
 
+/// Convert Duration to nanoseconds.
+#[inline]
+pub fn duration_to_nanos(d: Duration) -> u64 {
+    let nanos = d.subsec_nanos() as u64;
+    // Most of case, we can't have so large Duration, so here just panic if overflow now.
```

Comment:
```py
2 ** 64 / (1000000000 * 3600 * 365)
14038L
```

We may meet overflow after 1000 years later, I think we can ignore this check here.